File: Sequence/q1-q99/q60-q69/q61/q61.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* rotateRight(ListNode* head, int k) {
        if (k == 0 || head == NULL)
            return head;
        int len = 1;
        ListNode* tail = head;
        while (tail->next != NULL){
            len++;
            tail = tail->next;
        }
        tail->next = head;
        for (int i = 0; i < len - k%len; i++){
            head = head->next;
            tail = tail->next;
        }
        tail->next = NULL;
        return head;
    }
};

File: Sequence/q1-q99/q60-q69/q61/q61.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def rotateRight(self, head: ListNode, k: int) -> ListNode:
        if head == None or k == 0:
            return head
        size = 1
        tail = head
        while tail.next != None:
            size += 1
            tail = tail.next
        tail.next = head
        for i in range(size - k%size):
            head = head.next
            tail = tail.next
        tail.next = None
        return head
